syntax = 'proto3';

service FTQueue{
    rpc qCreate (label) returns (queueid) {}
    rpc qPush (Push_message) returns (void) {}
    rpc qId (label) returns (queueid) {}
    rpc qPop (queueid) returns (Item) {}
    rpc qTop (queueid) returns (Item) {}
    rpc qSize (queueid) returns (QueLength) {}
    rpc qDestroy (label) returns (void) {}
}

message label {
    string value = 1;
}

message queueid {
    int32 id = 1;
}

message Push_message{
    int32 queue_id = 1;
    string value = 2;
}

message Item{
    string value = 1;
}

message QueLength{
    int32 length = 1;
}


message void {
}

service FTQueueDistributed{
    rpc qCreateDistributed (label) returns (void) {}
    rpc qPushDistributed (Push_message) returns (void) {}
    rpc qIdDistributed (label) returns (void) {}
    rpc qPopDistributed (queueid) returns (void) {}
    rpc qTopDistributed (queueid) returns (void) {}
    rpc qSizeDistributed (queueid) returns (void) {}
    rpc qDestroyDistributed (label) returns (void) {}
}

message label_Dis {
    string value = 1;
    int32 SequenceNumber = 2;
}

message queueid_Dis {
    int32 id = 1;
    int32 SequenceNumber = 2;
}

message Push_message_Dis{
    int32 queue_id = 1;
    string value = 2;
    int32 SequenceNumber = 3;
}

message Item_Dis{
    string value = 1;
    int32 SequenceNumber = 2;
}

message QueLength_Dis{
    int32 length = 1;
    int32 SequenceNumber = 2;
}


message void_Dis {
}


